### PR TITLE
feat(mongodb): Update to SDK V4

### PR DESCRIPTION
**Pull request by yevgenypats**

### CODE REVIEW

File: plugins/destination/mongodb/client/migrate.go [hunk 1]
```diff
@@ -2,111 +2,110 @@ package client
...
-func isIndexOptionsConflictError(err error) bool {
-	cmdErr, ok := err.(mongo.CommandError)
-	if !ok {
-		return false
-	}
-
-	// This is either "Index already exists with a different name: %s" error or due to uniqueness change
-	return cmdErr.Name == "IndexOptionsConflict"
-}
+// func (c *Client) migrateTable(ctx context.Context, table *schema.Table) error {
```
Comment (disq) [inline plugins/destination/mongodb/client/migrate.go:23]: indexing and index migration seem to be missing?

File: plugins/destination/mongodb/main.go [hunk 2]
```diff
@@ -1,17 +1,24 @@
...
 const (
 	sentryDSN = "https://f4351c889aae4dde9b87a916d44ce836@o1396617.ingest.sentry.io/4504374724657152"
 )
 
 func main() {
-	p := destination.NewPlugin("mongodb", plugin.Version, client.New, destination.WithManagedWriter())
-	serve.Destination(p, serve.WithDestinationSentryDSN(sentryDSN))
+	p := plugin.NewPlugin("mongodb", internalPlugin.Version, client.New)
+	if err := serve.Plugin(p, serve.WithPluginSentryDSN(sentryDSN), serve.WithDestinationV0V1Server()).Serve(context.Background()); err != nil {
+		fmt.Println(err)
```
Comment (disq) [inline plugins/destination/mongodb/main.go:21]: nit:
```suggestion
				log.Fatalf("failed to serve plugin: %v", err)
```
then no need for os.Exit(1) below.

File: plugins/destination/mongodb/client/write.go [hunk 3]
```diff
@@ -108,7 +110,7 @@ func (c *Client) transformRecords(table *schema.Table, records []arrow.Record) [
 
 func (c *Client) appendTableBatch(ctx context.Context, table *schema.Table, docuemnts []any) error {
 	tableName := table.Name
-	if _, err := c.client.Database(c.pluginSpec.Database).Collection(tableName).InsertMany(ctx, docuemnts); err != nil {
+	if _, err := c.client.Database(c.spec.Database).Collection(tableName).InsertMany(ctx, docuemnts); err != nil {
```
Comment (candiduslynx) [inline plugins/destination/mongodb/client/write.go:113]:
```suggestion
	if _, err := c.client.Database(c.spec.Database).Collection(tableName).InsertMany(ctx, documents); err != nil {
```

File: plugins/destination/mongodb/client/write.go [hunk 4]
```diff
@@ -108,7 +110,7 @@ func (c *Client) transformRecords(table *schema.Table, records []arrow.Record) [
 
 func (c *Client) appendTableBatch(ctx context.Context, table *schema.Table, docuemnts []any) error {
```
Comment (candiduslynx) [inline plugins/destination/mongodb/client/write.go:111]:
```suggestion
func (c *Client) appendTableBatch(ctx context.Context, table *schema.Table, documents []any) error {
```

**Pull request merged**